### PR TITLE
No default workspace

### LINKED ISSUE #2316: Assign default-workspace if not set
When we pull the user object:

IF
- the default workspace id does not correspond to any workspace we have locally/is 0/is null(possible?)
- and we have any workspaces locally
THEN
- set (any) one of those workspaces as default workspace
- mark the user object as 'needs-sync'
- persist the user object
ELSE
- persist the user object like before

This will prevent the app from crashing in cases where the default workspace is not assigned correctly, which is a realistic case that can happen (and might be happening - for all we know this might be the cause of https://github.com/toggl/mobileapp/issues/2103 🤔 

I'm not sure where this should be implemented, but I think inside a special pull/persist sync state makes the most sense. We don't want to have logic like this in realm, and I wouldn't know where else to put it. But maybe there is a better way.

In either case, I never want it to persist the fetched user object before stepping through the above logic. I don't even want a small chance at getting crashes if we can fix it 'this easily' ourselves.

Note that (I suppose) it will still not persist anything if the local user object has pending changes? I'm not sure exactly how that will interact. Please post some details on that and the different edge cases below or in the PR.

### PR DESCRIPTION
**Pull request by simonrozsival**
Closes #2316 

I recommend reviewing commit by commit.

Integration tests: https://www.bitrise.io/build/07ae87ebd104bbb4

!! The DB schema version was changed 2 -> 3. This will be a bit of a problem when we merge it into `develop` - we need to update the schema to 4 when merging. !!

The owner of the hotfix branch can squash whenever

### CODE REVIEW

File: Toggl.Foundation/Interactors/Workspace/GetDefaultWorkspaceInteractor.cs [hunk 1]
```diff
@@ -20,7 +21,13 @@ public GetDefaultWorkspaceInteractor(ITogglDatabase database)
...
             => database.User
                 .Single()
-                .SelectMany(user =>
-                    database.Workspaces.GetById(user.DefaultWorkspaceId));
+                .SelectMany(user => user.DefaultWorkspaceId.HasValue
+                    ? database.Workspaces.GetById(user.DefaultWorkspaceId.Value)
+                    : chooseAnyWorkspace())
+                .Catch((InvalidOperationException exception) => chooseAnyWorkspace());
+
+        private IObservable<IDatabaseWorkspace> chooseAnyWorkspace()
```
Comment (juanlaube) [inline Toggl.Foundation/Interactors/Workspace/GetDefaultWorkspaceInteractor.cs]: this method name could be `chooseFirstWorkspace` or something similar, `chooseAnyWorkspace` sounds too random for me 😃

**Comment by simonrozsival** [comment]
The integration tests fail overall, but the test added in this PR passes (`00:00:18.3392640 - UserApiTests: The update method does not change default workspace when the value is null`).

File: Toggl.Foundation.MvvmCross/ViewModels/ReportsViewModel.cs [hunk 2]
```diff
@@ -107,13 +112,13 @@ public override void Prepare(long parameter)
             workspaceId = parameter;
         }
 
-        public async override Task Initialize()
+        public override async Task Initialize()
         {
             if (workspaceId == 0)
```
[the remark this answers is not in the document]
Comment (simonrozsival) [reply Toggl.Foundation.MvvmCross/ViewModels/ReportsViewModel.cs]: I believe that this can happen if you don't navigate to the view model with a parameter. @willsb can confirm this or he can correct me if I am wrong.
